Makefile

```diff
@@ -18,3 +18,6 @@ cppcheck: cppcheck-build-dir
 infer:
 	python3 setup.py build_ext -if -U NDEBUG | compiledb
 	@infer --fail-on-bug --compilation-database compile_commands.json
+
+valgrind:
+	PYTHONMALLOC=malloc valgrind --suppressions=hpy/tools/valgrind/python.supp --suppressions=hpy/tools/valgrind/hpy.supp --leak-check=full --show-leak-kinds=definite,indirect --log-file=/tmp/valgrind-output python -m pytest --valgrind --valgrind-log=/tmp/valgrind-output test/
```

Comment:
@thedrow Should we add `--gen-suppressions=all` here? It makes it convenient to write new suppressions or to compare with existing ones, but not sure what's considered best practice.

Reply:
We could be it is very verbose.
Instead, when we encounter a memory leak and verify it is a false positive, we can run the specific test that triggers it and generate a suppression for that.

The suppressions I've generated so far appears to be either genuine leaks from Python (one of them appears to be fixed in Python 3.9) or genuine memory leaks in HPy that we're aware of and won't fix for now.